src/alphabets/mod.rs

```diff
@@ -152,6 +152,72 @@ impl Alphabet {
     pub fn is_empty(&self) -> bool {
         self.symbols.is_empty()
     }
+
+    /// Return a new alphabet taking the intersect between this and other.
+    ///
+    /// # Example
+    /// ```
+    /// use bio::alphabets;
+    ///
+    /// let alpha_a = alphabets::Alphabet::new(b"acgtACGT");
+    /// let alpha_b = alphabets::Alphabet::new(b"atcgMVP");
+    /// let intersect_alpha = alpha_a.intersection(alpha_b);
+    ///
+    /// assert_eq!(intersect_alpha, alphabets::Alphabet::new(b"atcg"));
+    /// ```
+    pub fn intersection(self, other: Alphabet) -> Self {
+        let mut new_symbols = BitSet::new();
+        for symbol_to_add in self.symbols.intersection(&other.symbols) {
+            new_symbols.insert(symbol_to_add);
+        }
+        return Alphabet {
+            symbols: new_symbols,
+        };
```

Comment:
I think this can be simplified to
```suggestion
        Alphabet {
            symbols: self.symbols.intersection(&other.symbols).collect(),
        }
```

Reply:
Or perhaps it might also be possible to rephrase this in terms of `Alphabet::new(intersection stuff)`?

Reply:
Thanks -- made the updates in 01cf874. Learned something :)... I tried rephrasing with `new` but ran into some type inference issues I don't have facilities to solve yet.